### ISSUE TITLE
[Question/request] Possible for resourcepacks to change inventory background?

### ISSUE THREAD

**Issue by supersaiyansubtlety** [CottonMC/LibGui#179]
**Is your feature request related to a problem? Please describe.**
A user asked if my GUI background can be retextured. Since it's generated, I'm not sure.

**Describe the solution you'd like**
A way for resourcepacks to retexture GUI elements (if none exists). 

**Describe alternatives you've considered**
N/A

**Additional context**
I create my GUI here: https://gitlab.com/supersaiyansubtlety/automated_crafting/-/blob/master/src/main/java/net/sssubtlety/automated_crafting/gui/AutoCrafterGuiDescription.java#L59

**Comment by supersaiyansubtlety** [comment]
To clarify and distinguish this from #128, my preferred solution to this would allow users to assign complete textures to custom GUIs, not replace individual elements like slots and backgrounds for all LibGUI elements.

Perhaps allow devs to implement an interface on their GUI classes. The interface would have a method that returns a path to a texture that, if found, would be used instead of generated/assembled textures.

**Comment by Juuxel** [comment]
To allow this in your own gui, you can set a custom background painter for the root panel with a different texture path than LibGui's default (see how `BackgroundPainter.VANILLA` is defined).

**Comment by supersaiyansubtlety** [comment]
Is it possible to use generated textures if no custom texture is present?

**Comment by Juuxel** [comment]
Not really, but you can just copy LibGui's panel textures. Since they're just simple 9patch textures I made from vanilla's background colours, they're completely free to use.
